.gitignore

```diff
@@ -1,2 +1,7 @@
 /dev
 /.devcontainer
+output/
+cache/
+models/
```

Comment:
That's a great point. I normally add `task init` to ensure the directories are created during setup, and always exclude any artifact directories in my gitignore to avoid accidentally trying to push anything like this.

If you end up liking the Taskfile, then I'd just move the `mkdir` commands to the init, and have them created so not initialized by docker.

Also... 👏 thank you for clarifying the root user behavior as I've run into that and since I normally precreated artifact directories, I didn't realize that was the root cause. Nice to know!

Reply:
Also, the main reason I recommend something like this is because each project is able to reuse much of the same stuff. I actually moved to something different from Taskfile (Go based), but still think it's a great alternative to Makefile. 


Regardless, if you have a different perspective, totally fine if you don't want to merge this, just thought I'd offer. It could eliminate any readme driven setup and just have things ready to go. 

Thanks again for this great project. It's been a blast to experiment with! 👏

Reply:
I think this is a great tool, and I will definitely tell my colleagues at work about it!

Reply:
Nice! I'm glad this was useful. It's a great way to eliminate duplication and just bootstrap a project quickly. Here's a few examples I saved for jump starting things on some projects: [ci-configuration files - taskfiles](https://github.com/sheldonhull/ci-configuration-files/tree/main/taskfiles)

However, if you are in a different stack, I encourage you to consider the tooling in that stack. 

For example, InvokeBuild is far more powerful if you are comfortable with Powershell (and works cross platform). If in Python they have task runners that let you stick with Python.
I'm working in Go, so I work with Mage ([magetools](https://github.com/sheldonhull/magetools))

Taskfile is great for simple stuff, but if you end up needing something more complex with looping, parsing, etc then picking a more robust runner in your toolchain might benefit you. Keep that in mind and feel free to ask any follow-up on twitter or such. Always love helping out. Cheers!